tests/integration/hooks.rs

```diff
@@ -20,6 +20,12 @@ fn it_runs_scripts() {
             file::rename("POST-TEST", "POST");
         "#},
         )
+        .file(
```

Comment:
In general: more testing is needed.

At least check the behavior with and without `allow_commands`, as well as in the 'silent' mode (can't remember the flag). Should also test with failing commands.

Reply:
I added some tests, including one for silent mode and for a failing command :)

However, I'm not sure how to test the prompt - is there any way to do this? I've tried spawning the command as a subprocess and writing stuff to stdin, but it doesn't seem to work. I'm not sure if such a test would be strictly necessary but it would be nice.

Reply:
Promts are tricky... don't think I ever solved that one either :-)

@sassman, do we have a pattern for this somewhere?

Reply:
it's not really possible in an e2e test. I would address this with a very specific unit test, that makes sure that a certain function would lead to a prompt. 
For instance, with a dedicated enum variant as a return value, where the caller would then make sure to turn that value into a real prompt to the user. 
Right now we don't have such a thing yet, and prompts are quite poorly tested.